docsk8s/quick-start/ls-k8s-configuration-files.asciidoc

```diff
@@ -0,0 +1,284 @@
+--
+
+Pipelines can also be defined in separate files, as with the single pipeline example
+[source,yaml]
+--
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: pipeline2
+  labels:
+    app: logstash-demo
+data:
+  logstash-input.conf: |
+    input {
+      beats {
+        port => "5044"
+      }
+    }
+  logstash-output.conf: |
+    output {
+      elasticsearch {
+        hosts => ["https://demo-es-http:9200"]
+        index => "kube-apiserver-%{+YYYY.MM.dd}"
+        cacert => "/usr/share/logstash/config/es_ca.crt"
+        user => 'elastic'
+        password => '${ELASTICSEARCH_PASSWORD}'
+      }
+    }
+--
+
+Create the volume in your `Deployment`/`StatefulSet`
+
+[source,yaml]
+--
+volumes:
+  - name: logstash-pipelines-yaml
+    configMap:
+      name: logstash-pipelines-yaml
+  - name: pipeline2
+    configMap:
+      name: pipeline2
+--
+
+and mount the volume in your container spec
+
+[source,yaml]
+--
+#
+volumeMounts:
+    - name: pipeline2
+      mountPath: /usr/share/logstash/pipeline2
+    - name: logstash-pipelines-yaml
+      mountPath: /usr/share/logstash/config/pipelines.yml
+      subPath: pipelines.yml
+
+--
+
+[float]
+[[qs-settings]]
+
+==== Settings configuration
+
+[float]
+[[qs-logstash-yaml]]
+==== logstash.yml
+
+Unless a configuration file is specified, the default values for logstash.yml <insert link to logstash-settings-file> will be used. To override the default values, create a `ConfigMap` with the settings that you wish to override:
+
+[source,yaml]
+--
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: logstash-config
+  labels:
+    app: logstash-demo
+data:
+  logstash.yml: |
+    api.http.host: "0.0.0.0"
+    log.level: info
+    pipeline.workers: 2
+--
+
+In your `Deployment`/`StatefulSet`, create the `Volume`
+
+[source,yaml]
+--
+volumes:
+  - name: logstash-config
+    configMap:
+      name: logstash-config
+--
+
+Create the `volumeMount` in the `container`
+
+[source,yaml]
+--
+  volumeMounts:
+    - name: logstash-config
+      mountPath: /usr/share/logstash/config/logstash.yml
+      subPath: logstash.yml
+--
+
+
+[float]
+[[qs-jvm-options]]
+==== JVM Options
+
+JVM settings are best set using environment variables to override the default settings in `jvm.options`. This ensures that the expected settings from `jvm.options` are set, and only those options that explicitly need to be overriden are.
+
+The JVM settings should be added in the `LS_JAVA_OPTS` environment variable in the container definition of your `Deployment`/`StatefulSet`:
+
+[source,yaml]
+--
+spec:
+  containers:
+    - name: logstash
+      env:
+        - name: LS_JAVA_OPTS
+          value: "-Xmx2g -Xms2g"
+--
+
+[float]
+[[qs-logging]]
+==== Logging Configuration
+
+By default, we use the `log4j2.properties` from the logstash docker image, that will log to `stdout` only. To change the log level, to use debug logging, use the `log.level` option in <<qs-logstash-yaml, logstash.yml>>
```

Comment:
```suggestion
By default, we use the `log4j2.properties` file from the {ls} Docker image, which logs to `stdout` only. To change the log level, to use debug logging, use the `log.level` option in <<qs-logstash-yaml, logstash.yml>>.
```

Reply:
It's kinda weird that a link called `logstash-yml` links takes you to the topic in the LSR sometimes and to a topic in this document sometimes.  Let's discuss how to demystify that behavior for the user to provide more context.

Reply:
There are two different sets of `logstash.yml` content that I'd like to link to - one is how to define it in the context of kubernetes, and the other is the original `logstash.yml` docs, that show the settings. Maybe we don't need to do both, or maybe I'm misunderstanding the question...

Reply:
It's OK to link to both if we need to, but we need to disambiguate.  We don't want users to click links with identical link text, and land in two completely different books.